File: go/2018/day1/day1.go
package main

import (
	"fmt"
	"os"

	"../../common"
)

func part1(inputx interface{}, sum int) {
	freqs := make(map[int]bool)
	switch t := inputx.(type) {
	case []int:
		for {
			for _, val := range t {
				sum += val
				_, ok := freqs[sum]
				if ok {
					fmt.Printf("repeated:%d\n", sum)
					os.Exit(0)
				}
				freqs[sum] = true
			}
			fmt.Printf("sum = %d\n", sum)
		}
	}
}

func main() {
	ir := &common.InputReader{
		Filename:      "input.txt",
		TypeToConvert: "int",
	}
	x, _ := ir.ReadPuzzleInput()
	part1(x, 0)
}

File: python/2018/day1/day1.py

def main():
    s = 0
    freqs = []
    while 1:
        for line in open("input.txt","r"):
            s += int(line)
            if s in freqs:
                print("repeated",s)
                exit(0)
            freqs.append(s)
        print("sum",s)

if __name__ == "__main__":
    main()